can't modify unresolved.dependencies

Starting from version **0.43.0** I can no longer modify the value of the `unresolved.dependencies` property.  
Example: https://github.com/jonathanlermitage/intellij-extra-icons-plugin/blob/master/build.gradle.kts#L178-L187  
```kotlin
withType<DependencyUpdatesTask> {
        outputFormatter = closureOf<Result> {
            unresolved.dependencies.removeIf {
                val coordinates = "${it.group}:${it.name}"
                coordinates.startsWith("unzipped.com") || coordinates.startsWith("com.jetbrains:ideaI")
            }
            val plainTextReporter = PlainTextReporter(project, revision, gradleReleaseChannel)
            val writer = StringWriter()
            plainTextReporter.write(writer, this)
            logger.quiet(writer.toString().trim())
        }
}
```
(I'm doing this because Gradle can't find some artifacts version, like com.jetbrains:ideaIU. This is not a blocker issue, this is purely cosmetic. I'm curious... ^_^)

Actually, Gradle's error is weird: I ran `gradlew dependencyUpdates` and I got this error:
```
./gradlew dependencyUpdates --warning-mode all

> Configure project :
e: /home/jon/prj/intellij-extra-icons-plugin/build.gradle.kts:179:37: Unresolved reference: removeIf
e: /home/jon/prj/intellij-extra-icons-plugin/build.gradle.kts:180:38: Unresolved reference: it
e: /home/jon/prj/intellij-extra-icons-plugin/build.gradle.kts:180:50: Unresolved reference: it
e: /home/jon/prj/intellij-extra-icons-plugin/build.gradle.kts:185:37: Type mismatch: inferred type is StringWriter but OutputStream was expected

FAILURE: Build failed with an exception.

* Where:
Build file '/home/jon/prj/intellij-extra-icons-plugin/build.gradle.kts' line: 179

* What went wrong:
Script compilation errors:

  Line 179:             unresolved.dependencies.removeIf {
                                                ^ Unresolved reference: removeIf

  Line 180:                 val coordinates = "${it.group}:${it.name}"
                                                 ^ Unresolved reference: it

  Line 180:                 val coordinates = "${it.group}:${it.name}"
                                                             ^ Unresolved reference: it

  Line 185:             plainTextReporter.write(writer, this)
                                                ^ Type mismatch: inferred type is StringWriter but OutputStream was expected

4 errors

```
Line 179 refers to `unresolved.dependencies.removeIf`.
It makes no sens. This is a SortedSet (I checked this with a debugger), so I should be able to invoke `removeIf`. I also tried to invoke `clear()`  and I faced the same issue. But it seems to be a Set, because I can iterate over this set with `foreach`.

Maybe this is a side effect of the recent Kotlin migration? Some Kotlin basic collections are immutable. The debugger says this is a SortedSet from java.util, but I already observed some bugs with JDK17 and Kotlin: some lists from java.util were silently replaced by their equivalent class from Kotlin standard lib; at least for methods which declares a List return type but actually returns a java.util.ArrayList. It made me crazy. I finally fixed this by replacing the java.util.list return type by java.util.ArrayList, and replacing star imports of java.util by single-class imports, otherwise Kotlin seemed to pick the ArrayList from the Kotlin standard lib instead of java.util.*. It was super weird because it did not fail at compile-time, but at runtime, with Java17 only (no problem with Java11): _ArrayList from Kotlin std lib couldn't be casted to java.util.ArrayList_ ^_^.

--- a/gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/reporter/result/DependenciesGroup.kt
+++ b/gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/reporter/result/DependenciesGroup.kt
@@ -8,5 +8,5 @@ package com.github.benmanes.gradle.versions.reporter.result
  */
 data class DependenciesGroup<T : Dependency>(
   val count: Int,
-  val dependencies: Set<T> = sortedSetOf(),
+  val dependencies: MutableSet<T> = mutableSetOf(),
 )

--- a/gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/DependencyUpdatesReporter.kt
+++ b/gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/DependencyUpdatesReporter.kt
@@ -176,38 +176,38 @@ class DependencyUpdatesReporter(
     )
   }
 
-  private fun buildCurrentGroup(): Set<Dependency> {
+  private fun buildCurrentGroup(): MutableSet<Dependency> {
     return sortByGroupAndName(upToDateVersions)
       .map { dep ->
         updateKey(dep.key as HashMap)
         buildDependency(dep.value, dep.key)
-      }.toSortedSet() as TreeSet<Dependency>
+      }.toSortedSet()
   }
 
-  private fun buildOutdatedGroup(): Set<DependencyOutdated> {
+  private fun buildOutdatedGroup(): MutableSet<DependencyOutdated> {
     return sortByGroupAndName(upgradeVersions)
       .map { dep ->
         updateKey(dep.key as HashMap)
         buildOutdatedDependency(dep.value, dep.key)
-      }.toSortedSet() as TreeSet<DependencyOutdated>
+      }.toSortedSet()
   }
 
-  private fun buildExceededGroup(): Set<DependencyLatest> {
+  private fun buildExceededGroup(): MutableSet<DependencyLatest> {
     return sortByGroupAndName(downgradeVersions)
       .map { dep ->
         updateKey(dep.key as HashMap)
         buildExceededDependency(dep.value, dep.key)
-      }.toSortedSet() as TreeSet<DependencyLatest>
+      }.toSortedSet()
   }
 
-  private fun buildUndeclaredGroup(): Set<Dependency> {
+  private fun buildUndeclaredGroup(): MutableSet<Dependency> {
     return undeclared
       .map { coordinate ->
         Dependency(coordinate.groupId, coordinate.artifactId)
-      }.toSortedSet() as TreeSet<Dependency>
+      }.toSortedSet()
   }
 
-  private fun buildUnresolvedGroup(): Set<DependencyUnresolved> {
+  private fun buildUnresolvedGroup(): MutableSet<DependencyUnresolved> {
     return unresolved
       .sortedWith { a, b ->
         compareKeys(keyOf(a.selector), keyOf(b.selector))
@@ -311,7 +311,7 @@ class DependencyUpdatesReporter(
       )
     }
 
-    private fun <T : Dependency> buildDependenciesGroup(dependencies: Set<T>): DependenciesGroup<T> {
+    private fun <T : Dependency> buildDependenciesGroup(dependencies: MutableSet<T>): DependenciesGroup<T> {
       return DependenciesGroup<T>(dependencies.size, dependencies)
     }
 

--- a/gradle-versions-plugin/src/test/groovy/com/github/benmanes/gradle/versions/OutputFormatterSpec.groovy
+++ b/gradle-versions-plugin/src/test/groovy/com/github/benmanes/gradle/versions/OutputFormatterSpec.groovy
@@ -534,4 +534,92 @@ Failed to determine the latest version for the following dependencies (use --inf
     expected == actual
     result.task(':dependencyUpdates').outcome == SUCCESS
   }
+
+  def 'outputFormatter custom - modify unresolvable dependencies then outputs text output'() {
+    given:
+    buildFile = testProjectDir.newFile('build.gradle')
+    buildFile <<
+      """
+        import com.github.benmanes.gradle.versions.reporter.PlainTextReporter
+
+        buildscript {
+          dependencies {
+            classpath files($classpathString)
+          }
+        }
+
+        apply plugin: 'java'
+        apply plugin: 'com.github.ben-manes.versions'
+
+        repositories {
+          maven {
+            url '${mavenRepoUrl}'
+          }
+        }
+
+        dependencies {
+          implementation('backport-util-concurrent:backport-util-concurrent:3.1') { because 'I said so' }
+          implementation('backport-util-concurrent:backport-util-concurrent-java12:3.1')
+          implementation('com.google.guava:guava:99.0-SNAPSHOT') { because 'I know the future' }
+          implementation('com.google.guava:guava-tests:99.0-SNAPSHOT')
+          implementation('com.google.inject:guice:2.0') { because 'That\\'s just the way it is' }
+          implementation('com.google.inject.extensions:guice-multibindings:2.0')
+          implementation('com.github.ben-manes:unresolvable:1.0') { because 'Life is hard' }
+          implementation('com.github.ben-manes:unresolvable2:1.0')
+          implementation('com.github.ben-manes:unresolvable20:1.0')
+          implementation('com.github.ben-manes:unresolvable21:1.0')
+          implementation('com.github.ben-manes:unresolvable3:1.0')
+        }
+
+        dependencyUpdates {
+          outputFormatter = { result ->
+            result.unresolved.dependencies.removeIf {
+              it.name == 'unresolvable20' || it.name == 'unresolvable21'
+            }
+            def plainTextReporter = new PlainTextReporter(project, revision, gradleReleaseChannel)
+            plainTextReporter.write(System.out, result)
+          }
+          checkForGradleUpdate = false // future proof tests from breaking
+        }
+        """.stripIndent()
+
+    when:
+    def result = GradleRunner.create()
+      .withProjectDir(testProjectDir.root)
+      .withArguments('dependencyUpdates')
+      .withPluginClasspath()
+      .build()
+    def expected = """
+------------------------------------------------------------
+: Project Dependency Updates (report to plain text file)
+------------------------------------------------------------
+
+The following dependencies are using the latest milestone version:
+ - backport-util-concurrent:backport-util-concurrent:3.1
+     I said so
+ - backport-util-concurrent:backport-util-concurrent-java12:3.1
+
+The following dependencies exceed the version found at the milestone revision level:
+ - com.google.guava:guava [99.0-SNAPSHOT <- 16.0-rc1]
+     I know the future
+ - com.google.guava:guava-tests [99.0-SNAPSHOT <- 16.0-rc1]
+
+The following dependencies have later milestone versions:
+ - com.google.inject:guice [2.0 -> 3.1]
+     That's just the way it is
+     https://code.google.com/p/google-guice/
+ - com.google.inject.extensions:guice-multibindings [2.0 -> 3.0]
+     https://code.google.com/p/google-guice/
+
+Failed to determine the latest version for the following dependencies (use --info for details):
+ - com.github.ben-manes:unresolvable
+     Life is hard
+ - com.github.ben-manes:unresolvable2
+ - com.github.ben-manes:unresolvable3
+""".replace('\r', '').replace('\n', System.lineSeparator())
+
+    then:
+    assert result.output.toString().contains(expected)
+    result.task(':dependencyUpdates').outcome == SUCCESS
+  }
 }

